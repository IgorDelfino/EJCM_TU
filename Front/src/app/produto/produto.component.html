
<div class="row" >
    <p class="depart">Mais vendidos</p>
    <div class="column" *ngFor="let produto of produtos">
      <div class="card" style="width: 13rem;">
        <img class="card-img-top" [src]="[produto.img]" alt="Card image cap" style="width:100%; height: 30%;">
        <div class="card-body">
          <p class="card-description">{{produto.descricao}} </p>
          <p class="card-price" >{{produto.preco}}</p>
          <button type="button" class="button" (click)="mudaPreco(produto)">Comprar</button>
        </div>
      </div>
    </div>
