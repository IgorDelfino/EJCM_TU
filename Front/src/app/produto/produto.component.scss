.column {
    float: left;
    
    // /* padding: 0 5px; */
    padding: 0 15px;
  
  }

  .header{
    background-color: #238080;
    font-family: 'Qwigley', cursive;
    font-size: 25px;
    margin-left: 0;
    margin:0;
   }
   .titulo{
    margin:0px;
   }

  @import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@500&display=swap');
  /* Remove extra left and right margins, due to padding */
  .row {margin: 0 -5px;}

  /* Clear floats after the columns */
  .row:after {
    content: "";
    display: table;
    clear: both;
  }

  /* Responsive columns */
  @media screen and (max-width: 600x) {
    .column {
      width: 100%;
      display: block;
      margin-bottom: 20px;
      display:flex;
      justify-content: space-around;
      
    }
  }
  .card {
/* Add shadows to create the "card" effect */
      
      
      
      box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.25);
      border: 2px solid #FF4700;
      border-radius: 10px; /* 5px rounded corners */
      margin-top:10%;
      height: 50%;
     
      
  }
  .button{
      background-color: #FF4700;
      border: 1px solid #FF4700;
      color: white;
      padding: 8px;
      box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19);
      text-decoration: none;
      border-radius: 8px;
      /* margin-right:5%; */
     
      margin-left:27%;
      cursor: pointer;
      margin-bottom:2%;
      outline: none;
      

      
    
  }
  
  
  .card-body{
    margin-left: 6%;
    
  
  }
  .card-description{
    font-family: 'Montserrat', sans-serif;
    margin-top:0px;
    margin-left:0px;
  }
  .card-price{
      margin-top:-2%;
      font-weight: bolder;
      font-family: 'Montserrat', sans-serif;
  }
  .depart{
      font-size:20px;
      font-family: 'Montserrat', sans-serif;
      margin-bottom:-1%;
      
      margin-left:-2%;


  }
  .card-storage{
    font-family: 'Montserrat', sans-serif;
  }

  