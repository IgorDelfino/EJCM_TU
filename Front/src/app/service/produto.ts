export class Produto {
    descricao:string;
    preco:number;
    img:string;
    estoque: number;

}
