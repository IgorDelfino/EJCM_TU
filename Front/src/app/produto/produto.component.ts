import { Component, OnInit } from '@angular/core';
import { Produto } from '../service/produto';
// import { ProdutoService } from '../service/produto.service';

@Component({
  selector: 'app-produto',
  templateUrl: './produto.component.html',
  styleUrls: ['./produto.component.scss']
})
export class ProdutoComponent implements OnInit {
  //  Produtos:Produto[];
   produtos;
   a=1;
   b:number=2;
  constructor() { 
    // module.exports=this.somaItemsEstoque(this.produtos);

     
    this.produtos =[
      {
        descricao:"Bloco Adesivo Post-It® Rosa - 76 Mm X 76 Mm 1UN",
        preco:21,
        img:'../../assets/produtos/post-it.jpg',
        estoque: 30
        
      },
      {
        descricao:"Caderno Tilibra Capa Dura 80 Folhas Neon Laranja 1UN",
        preco:19,
        img:"../../assets/produtos/caderno.jpg",
        estoque: 20
      },
      {
        descricao:"Canetinha Hidrográfica 24 Cores Faber Castell 1UN ",
        preco:13,
        img:"../../assets/produtos/canetinha.jpg",
        estoque: 30
      },
      {
        descricao:"Papel Sulfite Chamequinho A-4 100 Folhas Cor Azul 1UN",
        preco:25,
        img:"../../assets/produtos/papel.jpg",
        estoque: 20
      },
     
      {
        descricao:"Caneta retrátil 1.0 BT C/2 TRION GRIP MOLIN 1200 1UN",
        preco:20,
        img:"../../assets/produtos/caneta.jpg",
        estoque: 30
      }
    ]
   
  }

  ngOnInit(): void {
    
  }
   
   alteraEstoque(produto){
    console.log(produto.estoque)
       produto.estoque --;
       console.log(produto.estoque)
       return;
    };
    Sum(a, b){
      return a + b;
    }
 
    forEach(items, callback){
      for(let index = 0; index < items.length; index++){
        callback(items[index]);
      }
    }
  // mudaPreco(produto){
  //     console.log(produto.preco)
  //     if ((produto.preco >= 20)){
  //        produto.preco ++;
  //        console.log(produto.preco)
  //        return;
  //     }
  // somaItemsEstoque(produtos){
  //   return produtos(function(total, produto){
  //     return total + produto.estoque;

  //   }, 0);
  // };
  
}


// quando fazemos uma função no component.ts, precisamos mudar no html e no produto.ts e também no component.spec.ts