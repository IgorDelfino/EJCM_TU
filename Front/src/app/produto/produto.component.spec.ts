import { async, ComponentFixture } from '@angular/core/testing';
import { ConfigureFn, configureTests } from 'src/test-config.helper'
import { NO_ERRORS_SCHEMA, ÉµivyEnabled as ivyEnabled, Component } from '@angular/core';
import { ProdutoComponent } from './produto.component';






describe('<ProdutoComponent /> ', () => {
    let component: ProdutoComponent;
    let fixture: ComponentFixture<ProdutoComponent>;
  
    beforeEach(async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [ProdutoComponent],
          schemas: [NO_ERRORS_SCHEMA]
        });
      };
  
       configureTests(configure).then(testBed => {
         fixture = testBed.createComponent(ProdutoComponent);
         component = fixture.componentInstance;
         fixture.detectChanges();
      });
     }));
    
    
}) 