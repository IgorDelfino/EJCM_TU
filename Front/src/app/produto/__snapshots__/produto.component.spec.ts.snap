// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProdutoComponent />  #render should generate snapshot for ProdutoComponent 1`] = `
<app-produto
  a={[Function Number]}
  b={[Function Number]}
  produtos={[Function Array]}
  toggleCompra={[Function Boolean]}
>
  <div
    class="row"
  >
    <p
      class="depart"
    >
      Mais vendidos
    </p>
    
    <div
      class="column"
    >
      <div
        class="card"
        style="width: 13rem;"
      >
        <img
          alt="Card image cap"
          class="card-img-top"
          src="../../assets/produtos/post-it.jpg"
          style="width:100%; height: 30%;"
        />
        <div
          class="card-body"
        >
          <p
            class="card-description"
          >
            Bloco Adesivo Post-It® Rosa - 76 Mm X 76 Mm 1UN 
          </p>
          <p
            class="card-price"
          >
            Preço: 21
          </p>
          <p
            class="card-storage"
          >
            Estoque: 30
          </p>
          <button
            class="button"
            type="button"
          >
            Comprar
          </button>
        </div>
      </div>
    </div>
    <div
      class="column"
    >
      <div
        class="card"
        style="width: 13rem;"
      >
        <img
          alt="Card image cap"
          class="card-img-top"
          src="../../assets/produtos/caderno.jpg"
          style="width:100%; height: 30%;"
        />
        <div
          class="card-body"
        >
          <p
            class="card-description"
          >
            Caderno Tilibra Capa Dura 80 Folhas Neon Laranja 1UN 
          </p>
          <p
            class="card-price"
          >
            Preço: 19
          </p>
          <p
            class="card-storage"
          >
            Estoque: 20
          </p>
          <button
            class="button"
            type="button"
          >
            Comprar
          </button>
        </div>
      </div>
    </div>
    <div
      class="column"
    >
      <div
        class="card"
        style="width: 13rem;"
      >
        <img
          alt="Card image cap"
          class="card-img-top"
          src="../../assets/produtos/canetinha.jpg"
          style="width:100%; height: 30%;"
        />
        <div
          class="card-body"
        >
          <p
            class="card-description"
          >
            Canetinha Hidrográfica 24 Cores Faber Castell 1UN  
          </p>
          <p
            class="card-price"
          >
            Preço: 13
          </p>
          <p
            class="card-storage"
          >
            Estoque: 30
          </p>
          <button
            class="button"
            type="button"
          >
            Comprar
          </button>
        </div>
      </div>
    </div>
    <div
      class="column"
    >
      <div
        class="card"
        style="width: 13rem;"
      >
        <img
          alt="Card image cap"
          class="card-img-top"
          src="../../assets/produtos/papel.jpg"
          style="width:100%; height: 30%;"
        />
        <div
          class="card-body"
        >
          <p
            class="card-description"
          >
            Papel Sulfite Chamequinho A-4 100 Folhas Cor Azul 1UN 
          </p>
          <p
            class="card-price"
          >
            Preço: 25
          </p>
          <p
            class="card-storage"
          >
            Estoque: 20
          </p>
          <button
            class="button"
            type="button"
          >
            Comprar
          </button>
        </div>
      </div>
    </div>
    <div
      class="column"
    >
      <div
        class="card"
        style="width: 13rem;"
      >
        <img
          alt="Card image cap"
          class="card-img-top"
          src="../../assets/produtos/caneta.jpg"
          style="width:100%; height: 30%;"
        />
        <div
          class="card-body"
        >
          <p
            class="card-description"
          >
            Caneta retrátil 1.0 BT C/2 TRION GRIP MOLIN 1200 1UN 
          </p>
          <p
            class="card-price"
          >
            Preço: 20
          </p>
          <p
            class="card-storage"
          >
            Estoque: 30
          </p>
          <button
            class="button"
            type="button"
          >
            Comprar
          </button>
        </div>
      </div>
    </div>
  </div>
</app-produto>
`;
